query/src/main/java/io/keen/client/java/Query.java

```diff
@@ -151,38 +181,61 @@ protected Query(Builder builder) {
      * individual queries may require additional arguments.
      */
     public static class Builder {
+        // required
         private QueryType queryType;
         private String eventCollection;
+        private Timeframe timeframe;
+
+        // sometimes optional
         private String targetProperty;
 
         // required by the Percentile query
         private Double percentile;    // 0-100 with two decimal places of precision for example, 99.99
 
         // optional
-        private Timeframe timeframe;
-        private List<Map<String, Object>> filters;
+        private Collection<Filter> filters;
         private String interval;
-        private String timezone;
-        private ArrayList<String> groupBy;
+        private List<String> groupBy;
         private Integer maxAge;
 
         public Builder(QueryType queryType) {
             this.queryType = queryType;
         }
 
         /**
-         * get filters
-         * @return a list of filters.
+         * Get filters
+         *
+         * @return a collection of filters.
+         */
+        public Collection<Filter> getFilters() { return filters; }
+
+        /**
+         * Set the list of filters. Existing filters will be discarded.
+         *
+         * @param filters the filter arguments.
          */
-        public List<Map<String, Object>> getFilters() { return filters; }
+        public void setFilters(Collection<? extends Filter> filters) {
+            this.filters = null;
+
+            // Client code may just be clearing all the filters.
+            if (null != filters) {
+                withFilters(filters);  // Shallow copy the list of steps
+            }
+        }
 
         /**
-         * set filters
-         * @param filters  the filter arguments.
+         * Adds the given filters as optional parameters to the query.
+         *
+         * @param filters the filter arguments.
+         * @return This instance (for method chaining).
          */
-        public void setFilters(List<Map<String, Object>> filters) {this.filters = filters;}
-        public Builder withFilters(List<Map<String, Object>> filters) {
-            setFilters(filters);
+        public Builder withFilters(Collection<? extends Filter> filters) {
```

Comment:
So the issues is two-fold, here:
1) ```withFilters()``` now doesn't destroy existing filters
2) ```Collection<Filter-ish>``` as the parameter

In the second case, I'd say we can still be appropriately covariant and accept any Collection if we want to, and create a List from it...which is a step toward expressing filters as an unordered bag instead of a sequence. That shouldn't break existing code as long as T is  Map<String, Object> I believe. Thoughts?